fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt

```diff
@@ -425,8 +449,16 @@ private enum class Highlight(
 private fun ProductRecommendation(
     product: RecommendedProductState.Product,
     onClick: () -> Unit,
+    onImpression: (RecommendedProductState.Product) -> Unit,
```

Comment:
Could be (String) -> Unit since we only need to pass the aid, same for all the functions calling into this

Reply:
True true. I was initially leaning towards passing the entire data model, so the UI isn't necessarily aware of the API requirements, especially if they were to change down the road. 

Another option for consideration could be passing back nothing and having the middleware read the state directly and obtain the `aid` that way.

Reply:
hmm I think just passing the id would be make sense, in case we start showing two recommendations here, middleware woulnd't be able to know without extra info which recommendations had an impression and click.

I think we should also add make `onClick` of type `(String) -> Unit`, since we need to know which recommendation was clicked as well.